### PR TITLE
pybind/mgr/stats: validate cmdtag

### PR DESCRIPTION
**Pull request by joscollin**
Fixes: https://tracker.ceph.com/issues/51975

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/pybind/mgr/stats/fs/perf_stats.py [hunk 1]
```diff
@@ -124,6 +124,13 @@ def set_client_metadata(self, client_id, key, meta):
             result[key] = meta
 
     def notify(self, cmdtag):
+        try:
+            uuid.UUID(str(cmdtag))
```
Comment (vshankar) [inline src/pybind/mgr/stats/fs/perf_stats.py:128]: You need to check if `cmdtag` is a valid key in `self.client_metadata['in_progress']`.

**Review by vshankar — CHANGES_REQUESTED** [review]

File: src/pybind/mgr/stats/fs/perf_stats.py [hunk 2]
```diff
@@ -126,7 +126,12 @@ def set_client_metadata(self, client_id, key, meta):
     def notify(self, cmdtag):
         self.log.debug("cmdtag={0}".format(cmdtag))
         with self.meta_lock:
-            result = self.client_metadata['in_progress'].pop(cmdtag)
+            try:
+                result = self.client_metadata['in_progress'].pop(cmdtag)
+            except KeyError:
+                self.log.warn("cmdtag not found in client metadata")
```
Comment (sebastian-philipp) [inline src/pybind/mgr/stats/fs/perf_stats.py:132]:
```suggestion
                self.log.warn(f"cmdtag {cmdtag}  not found in client metadata")
```

Comment (joscollin) [reply src/pybind/mgr/stats/fs/perf_stats.py:132]: We are **mostly** expecting empty values for `cmdtag` here. So this change doesn't make a difference.

Comment (joscollin) [reply src/pybind/mgr/stats/fs/perf_stats.py]: I've fixed it anyway. It works better if a uuid is not found, somehow. Thanks Seb.

**Review by sebastian-philipp — APPROVED** [review]

**Review by vshankar — APPROVED** [review]

**Comment by joscollin** [comment]
jenkins test signed

**Comment by joscollin** [comment]
jenkins test submodules

**Comment by joscollin** [comment]
http://pulpito.front.sepia.ceph.com/jcollin-2021-08-06_09:28:55-fs:functional:-wip-B51975-keyerror-null-foo-distro-basic-smithi/

**Pull request merged**